src/Collection/CollectionTrait.php

```diff
@@ -399,6 +399,67 @@ public function last()
         return $result;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function lastN($howMany)
+    {
+        if ($howMany < 1) {
+            throw new \InvalidArgumentException("lastN requires a number greater than 0");
+        }
+
+        $iterator = $this->optimizeUnwrap();
+        if (is_array($iterator)) {
+            return new Collection(array_slice($iterator, $howMany * -1));
+        }
+
+        if ($iterator instanceof Countable) {
+            $count = count($iterator);
+
+            if ($count === 0) {
+                return new Collection([]);
+            }
+
+            $iterator = new LimitIterator($iterator, max(0, $count - $howMany), $howMany);
+
+            return new Collection($iterator);
+        }
+
+        $generator = function ($iterator, $howMany) {
+            $result = [];
+            $bucket = 0;
+            $offset = 0;
+            $toggleOffset = false;
+
+            foreach ($iterator as $k => $item) {
```

Comment:
This loop is pretty dense are there any comments that could be added for to explain why results are being sliced into buckets?